browser/ui/views/brave_actions/brave_actions_container.cc

```diff
@@ -279,16 +232,25 @@ void BraveActionsContainer::RemoveAction(const std::string& id) {
   PreferredSizeChanged();
 }
 
-// Adds or removes action
-void BraveActionsContainer::ShowAction(const std::string& id, bool show) {
-  if (show != IsActionShown(id))
-    show ? AddAction(id) : RemoveAction(id);
+void BraveActionsContainer::UpdateActionVisibility(const std::string& id) {
+  if (views::Button* button = GetActionButton(id)) {
+    bool should_show = ShouldShowAction(id);
+    if (button->GetVisible() != should_show) {
+      button->SetVisible(should_show);
+      Update();
+      PreferredSizeChanged();
```

Comment:
This is redundant call. - When child view's visibility is changed, this container(parent) will do re-layout.

Reply:
Fixed.